src/sentry/integrations/slack/utils/feature.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from sentry.models.integrations.integration import Integration
+from sentry.services.hybrid_cloud.integration import RpcIntegration
+
+_default_logger = logging.getLogger(__name__)
+
+
+def organization_integration_has_feature_enabled(
+    integration: Integration | RpcIntegration, organization_id: int, feature: str
+) -> bool:
+    """
+    Helper function to check if an integration's installation has a specific feature flag
+    """
+    from sentry.integrations.slack.integration import SlackIntegration
+
+    base_logs = {
+        "organization_id": organization_id,
+        "integration_id": integration.id,
+    }
+    installation = integration.get_installation(organization_id=organization_id)
+    if installation is None:
+        _default_logger.info(  # type: ignore[unreachable]
+            "no installation found for integration",
+            extra=base_logs,
+        )
+        return False
+
+    if not isinstance(installation, SlackIntegration):
+        _default_logger.info(
+            "installation object is not a slack integration",
+            extra=base_logs,
+        )
+        return False
+
+    return installation.has_feature(feature_name=feature)
```

Comment:
Didn't spend a lot of time on it but I think this method can be more generic and used in parent integration class instead of being specific to slack. You need one wrapper around intallation.has_feature to make sure installation exists (super.organization_integration_has_feature_enabled) and then the integration specific method check to make sure the integration is slack.

Like instead of utils make these class methods of IntegrationFeatures and then have SlackFeatures inheriting it.

Reply:
Yeah, was thinking the same, let me see what I can do

Reply:
I'll add that in a follow up PR as to not block GA